src/index.js

```diff
@@ -94,45 +100,45 @@ const SplunkRum = {
-    const pluginConf = { ignoreUrls };
+    
+    // enabled: false prevents registerInstrumentations from enabling instrumentations in constructor
+    // they will be enabled in registerInstrumentations
+    const pluginConf = { ignoreUrls, enabled: false };
+    
     const provider = new PatchedWTP({
-      plugins: [
-        new SplunkDocumentLoad(),
+      logLevel: options.debug ? DiagLogLevel.DEBUG : DiagLogLevel.ERROR,
+    });
+
+    this._deregisterInstrumentations = registerInstrumentations({
+      tracerProvider: provider,
+      instrumentations: [
+        new SplunkDocumentLoad(pluginConf),
         new SplunkXhrPlugin(pluginConf),
-        new SplunkUserInteractionPlugin({ adjustAutoInstrumentedEvents }),
+        new SplunkUserInteractionInstrumentation(Object.assign({}, pluginConf, {adjustAutoInstrumentedEvents})),
+        new PostDocLoadResourceObserver(
+          options.allowedInitiatorTypes ?
+            Object.assign({}, pluginConf, { allowedInitiatorTypes: options.allowedInitiatorTypes }) :
+            pluginConf
+        ),
+        new SplunkFetchInstrumentation(pluginConf),
+        new SplunkWebSocketInstrumentation(pluginConf),
+        new SplunkLongTaskInstrumentation(pluginConf),
       ],
-      logLevel: options.debug ? LogLevel.DEBUG : LogLevel.ERROR,
     });
+
     if (options.spanProcessor) {
       const sp = options.spanProcessor;
       if (typeof sp.onStart === 'function' && typeof sp.onEnd === 'function') {
         provider.addSpanProcessor(sp);
       }
     }
-    
-    new WebSocketInstrumentation(provider, pluginConf).patch();
-    
-    const fetchInstrumentation = new SplunkFetchInstrumentation(pluginConf);
-    fetchInstrumentation.setTracerProvider(provider);
-    fetchInstrumentation.enable();
-
-    const longtaskInstrumentation = new SplunkLongTaskInstrumentation();
-    longtaskInstrumentation.setTracerProvider(provider);
-    
-    if (options.allowedInitiatorTypes) {
-      pluginConf.allowedInitiatorTypes = options.allowedInitiatorTypes;
-    }
-    new PostDocLoadResourceObserver(pluginConf).setTracerProvider(provider);
 
     if (options.beaconUrl) {
       const completeUrl = options.beaconUrl + (options.rumAuth ? '?auth='+options.rumAuth : '');
-      const bufferTimeout = typeof options.bufferTimeout !== 'undefined' 
-        ? Number(options.bufferTimeout)
-        : 5000; //millis, tradeoff between batching and loss of spans by not sending before page close
       const batchSpanProcessor = new BatchSpanProcessor(new PatchedZipkinExporter(completeUrl), {
-        // This number is still an experiment, probably not the final number yet.
-        bufferTimeout,
-        bufferSize: 20, // spans, tradeoff between batching and hitting sendBeacon invididual limits
+        scheduledDelayMillis: options.bufferTimeout,
+        _maxExportBatchSize: options.bufferSize, 
+        _maxQueueSize: 2 * options.bufferSize,
```

Comment:
already noted in slack, reminder

```suggestion
        maxExportBatchSize: options.bufferSize, 
        maxQueueSize: 2 * options.bufferSize,
```

Reply:
done